pkg/apis/pubsub/internal/deprecated_condition.go

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package internal
+
+import (
+	"reflect"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+const deprecationMessage = "This object is deprecated and should be deleted. There is no public " +
+	"replacement. If you want a public replacement, please comment on " +
+	"https://github.com/google/knative-gcp/issues/905. This object should be deleted before " +
```

Comment:
Content is correct. We could mellow down a bit. How about: "This object (or kind) is deprecated and the CRD is made 'internal'. This means, end-users should not create objects of this CRD directly. If you need a non-internal (or a public) variant, then please let us know by commenting on https://github.com/google/knative-gcp/issues/905. Moreover, the object must be deleted before upgrading to 0.16 to avoid leaking (or orphaning) the Google Cloud Platform resources that were created on-behalf of this object, such as Pub/Sub topics and subscriptions."

Reply:
Sounds good, how about (almost identical to yours, but removed the parenthesis):

`This Kind is deprecated and the CRD has been made 'internal'. This means, end-users should not create objects of this CRD directly. If you need a non-internal variant, then please let us know by commenting on https://github.com/google/knative-gcp/issues/905. Moreover, the object must be deleted before upgrading to 0.16 to avoid orphaning the Google Cloud Platform resources that were created on-behalf of this object, such as Pub/Sub Topics and Subscriptions.`

Reply:
/LGTM